src/views/webview-app/overview-page.tsx

```diff
@@ -55,30 +61,83 @@ const OverviewPage: React.FC = () => {
     resetGlobalCSS();
   }, []);
 
+  function handleOpenFileChooserResult<T>(): Promise<T> {
+    const requestId = handleOpenFileChooser();
+    return new Promise((resolve) => {
+      const messageHandler = (event) => {
+        const message: MESSAGE_FROM_EXTENSION_TO_WEBVIEW = event.data;
+        if (
+          message.command === MESSAGE_TYPES.OPEN_FILE_CHOOSER_RESULT &&
+          message.requestId === requestId
+        ) {
+          window.removeEventListener('message', messageHandler);
+          resolve(message.files as T);
+        }
+      };
+      window.addEventListener('message', messageHandler);
+    });
+  }
+
+  // Electron 32.0 removed support for the `path` property of the Web File object in favor of the webUtils.getPathForFile method.
+  // https://github.com/electron/electron/blob/83d704009687956fb4b69cb13ab03664d7950118/docs/breaking-changes.md%23removed-filepath
+  // We can not import `dialog` and `webUtils` from 'electron' in the sandboxed webview.
+  // To work around this, we use a custom dialog provider that uses webview APIs
+  // to send a message to the extension process to open the electron file dialog
+  // and listen for the response to get the file path and send them to the electron file input backend.
+  const dialogProvider = {
+    getCurrentWindow(): void {},
+    dialog: {
+      showSaveDialog(): Promise<{ canceled: boolean; filePath?: string }> {
+        return handleOpenFileChooserResult<Uri | undefined>().then(
+          (file?: Uri) => {
+            return { canceled: false, filePath: file?.path };
+          }
+        );
+      },
+      showOpenDialog(): Promise<{ canceled: boolean; filePaths: string[] }> {
+        return handleOpenFileChooserResult<Uri[] | undefined>().then(
+          (files?: Uri[]) => {
+            return {
+              canceled: false,
+              filePaths: files ? files?.map((file) => file.path) : [],
```

Comment:
Shouldn't this be `.fsPath` instead of `.path`?

Reply:
Looks like we don't have this value in the returned Uri object:
```
open files----------------------
extensionHostProcess.js:162
[
  Ko {
    scheme: 'file',
    authority: '',
    path: '/Users/alena.khineika/ca.pem',
    query: '',
    fragment: '',
    _formatted: null,
    _fsPath: null
  }
]
```

Reply:
@alenakhineika Yeah no, it's a getter on the class, that's why it doesn't show up in `util.inspect()`. You'll want to check that this actually works for Windows, and for files with characters that need URL escaping (e.g. spaces) in their paths.

Reply:
This output is actually printed with `util.inspect()`, and when I tried to replace it, the value was not present.

Reply:
The value is present if I print `util.inspect((files as vscode.Uri[]).map(file => file.fsPath))`, as expected (and as documented in the VSCode API)

Reply:
Yeah, i have no idea what i was printing last time 🙈 tried again and you are right, the value is there.